chain-time/src/timeline.rs

```diff
@@ -0,0 +1,37 @@
+use std::time::{Duration, SystemTime};
+
+/// Represent a timeline with a specific start point rooted on earth time.
+pub struct Timeline(pub(crate) SystemTime);
+
+/// Represent an offset in time units in the timeline
+pub struct TimeOffset(pub(crate) Duration);
+
+impl From<SystemTime> for Timeline {
+    fn from(s: SystemTime) -> Self {
+        Timeline(s)
+    }
+}
+
+impl Timeline {
+    /// Create a new timeline, which is a time starting point
+    pub fn new(start_time: SystemTime) -> Self {
+        Timeline(start_time)
+    }
+
+    /// Return the duration since the creation of the timeline
+    ///
+    /// If the time is earlier than the start of this timeline,
+    /// then None is returned.
+    pub fn differential(&self, t: &SystemTime) -> Option<TimeOffset> {
```

Comment:
I wanted to bikeshed this method name but then I found out it is [Doctor Who compliant](https://tardis.fandom.com/wiki/Time_differential). Approved.